src/ngraph/pass/reshape_elimination.cpp

```diff
@@ -192,3 +191,127 @@ void pass::ReshapeElimination::construct_dot_transpose_pattern()
     auto m = make_shared<pattern::Matcher>(preshape, callback);
     this->add_matcher(m);
 }
+
+void pass::RecurrentReshapeElimination::construct_recurrent_reshape()
+{
+    Shape shape_op{3};
+    Shape shape_r{1, 3};
+
+    auto no_fan_out = [](std::shared_ptr<Node> n) {
+        auto users = n->get_users(true);
+        std::set<std::shared_ptr<Node>> user_set(users.begin(), users.end());
+        size_t num_unique_users = user_set.size();
+        if (num_unique_users == 1)
+        {
+            return true;
+        }
+        else
+        {
+            NGRAPH_DEBUG << "recurrent_reshape_elimination: " << n->get_name() << " has fan out\n";
+            return false;
+        }
+    };
+
+    auto op = make_shared<pattern::op::Label>(element::f32, shape_op);
+    auto reshape = make_shared<op::Reshape>(op, AxisVector{0}, shape_r);
+    auto rehape_label = make_shared<pattern::op::Label>(reshape, no_fan_out, NodeVector{reshape});
+
+    auto callback = [op, rehape_label](pattern::RecurrentMatcher& m) {
+        NGRAPH_DEBUG << "In callback for construct_recurrent_reshape against node = "
+                     << rehape_label->get_argument(0)->get_name();
+        auto reshape_node_vector = m.get_bound_nodes_for_pattern(rehape_label);
+
+        // The bounded node vector is in reverse order. It is convinient to have the
+        // bounded node vector in the correct order
+        std::reverse(std::begin(reshape_node_vector), std::end(reshape_node_vector));
+
+        auto first_bounded_reshape_op = reshape_node_vector.front();
+        auto driver_op = first_bounded_reshape_op->get_argument(0);
+        auto last_bounded_reshape_op = reshape_node_vector.back();
+
+        // Need to check if the next to last bounded op is a reshape since the last reshape is allowed
```

Comment:
`next to last` -> `user of the last` (right?)

Reply:
Yes